### DISCUSSION TITLE
assumption of order

### DISCUSSION THREAD

**Discussion by pengguanya** [insightsengineering/rbmi#363] [General]
@gowerc 
It seems `pool_bootstrap_percentile` relies on the assumption of the order of the visits which were intrinsically determined by where they appeared in the drawed or imputated dataframes. How did it guarantee that data were initially ordered as expected for `pool_bootstrap_percentile`, the same for `pool_bootstrap_normal`
https://github.com/insightsengineering/rbmi/blob/a45df71eeff8b3a134523a5c73e47aa986953ff2/R/pool.R#L431-L433

**Comment by nociale** [comment]
Hi @pengguanya 

I think these functions you mention rely on the fact that the first element of `est` is the estimate from the original dataset (and not from a leave-one-out or a bootstrapped dataset). This is internally managed based on the user's choice of `method` (already at the call to `draws()`), and cannot be changed by the user. So, as long as there is not any internal error, the first estimate should always be referred to the estimate from the original dataset for these particular methods. Also `pool_internal.jackknife()` rely on the same assumption. I don't think it is necessary to check that the order of `est` is equivalent throughout the call to all the sequential functions from `draws()` to `pool()` (and I don't know how this could be done with the current implementation).

This is the line of code that first insert the estimate from the original dataset at the beginning of `samples` in `draws()` (which then is used to create `imputations` objects and `est`...):

```
if (first_sample_orig) {
        samples <- append(list(initial_sample), samples)
}
```
 `est[1]` is the treatment effect estimate derived from the analysis of the original dataset which was imputed based on a MMRM fit that produced `initial_sample` in `draws()`.
